### PR TITLE
Add option for PodDisruptionBudget

### PR DESCRIPTION
**Pull request by Mahpara93**
Add option for PodDisruptionBudget(pdb) to opentelemetry-collector

### CODE REVIEW

File: charts/opentelemetry-collector/values.yaml [hunk 1]
```diff
@@ -312,6 +312,11 @@ serviceMonitor:
   extraLabels: {}
   #  release: kube-prometheus-stack
 
+# PodDisruptionBudget (pdb) is used only if standaloneCollector enabled
+pdb:
```
Comment (dmitryax) [inline charts/opentelemetry-collector/values.yaml:316]: pdb may be too ambiguous in scope of values.yaml. Let's use full name

File: charts/opentelemetry-collector/values.yaml [hunk 2]
```diff
@@ -312,6 +312,11 @@ serviceMonitor:
...
+  enabled: false
+  minAvailable: 1
```
Comment (dmitryax) [inline charts/opentelemetry-collector/values.yaml:318]: let's add `maxUnavailable`

File: charts/opentelemetry-collector/values.yaml (hunk 1 above)
Comment (Mahpara93) [reply charts/opentelemetry-collector/values.yaml]: Good idea! I changed it to PodDisruptionBudget.

File: charts/opentelemetry-collector/values.yaml (hunk 2 above)
Comment (Mahpara93) [reply charts/opentelemetry-collector/values.yaml]: I added maxUnavailable as well.

File: charts/opentelemetry-collector/values.yaml [hunk 3]
```diff
@@ -312,6 +312,12 @@ serviceMonitor:
   extraLabels: {}
   #  release: kube-prometheus-stack
 
+# PodDisruptionBudget is used only if standaloneCollector enabled
+PodDisruptionBudget:
```
Comment (dmitryax) [inline charts/opentelemetry-collector/values.yaml:316]: Should not be capitalized `podDisruptionBudget` according to helm style guides

Comment (Mahpara93) [reply charts/opentelemetry-collector/values.yaml]: fixed

File: charts/opentelemetry-collector/templates/pdb.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{{- if .Values.podDisruptionBudget.enabled .Values.standaloneCollector.enabled }}
```
Comment (dmitryax) [inline charts/opentelemetry-collector/templates/pdb.yaml:1]: Hm.. I think `and` is missed here

File: charts/opentelemetry-collector/templates/pdb.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+{{- if .Values.podDisruptionBudget.enabled and .Values.standaloneCollector.enabled }}
```
Comment (dmitryax) [inline charts/opentelemetry-collector/templates/pdb.yaml:1]: This doesn't work :) Should be:
```suggestion
{{- if and .Values.podDisruptionBudget.enabled .Values.standaloneCollector.enabled }}
```

Comment (Mahpara93) [reply charts/opentelemetry-collector/templates/pdb.yaml]: Thanks! Appreciate your feedback :)

**Review by dmitryax — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**